kernel/src/drivers/ps2/mod.rs

```diff
@@ -90,167 +93,159 @@ impl Ps2Controller {
             println!("ps2c: detected no available devices");
         }
 
-        flush_output();
+        flush_output()?;
 
         self.initialized = true;
+
+        Ok(())
     }
 
     /// Initializes the config for this controller
-    fn initialize_config(&mut self) {
+    fn initialize_config(&mut self) -> Result<(), Ps2Error> {
         // Read the config from the controller
-        self.read_config();
+        self.read_config()?;
```

Comment:
Maybe the whole config reading/writing stuff could be extracted to read and return a config rather than mutating self?

Reply:
The config still needs to be mutably updated, though.

Reply:
We can set it through the field.

Reply:
I see. That's probably a cleaner way of doing it.